test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_cad.cxx

```diff
@@ -365,6 +367,191 @@ TEST (t8_gtest_geometry_cad, jacobian)
+  }
+}
+
+INSTANTIATE_TEST_SUITE_P (t8_gtest_check_2d_element_linear_cad_curve, class_2d_element_linear_cad_curve, AllEclasses2D);
+
+#endif /* T8_WITH_OCC */
+
+#if T8_WITH_OCC
+/* The test checks if the mapping algorithms for curved 2d elements shift values on a curved cad curve correctly. */
+class class_2d_element_curved_cad_curve: public testing::TestWithParam<t8_eclass_t> {
+ protected:
+  void
+  SetUp () override
+  {
+    eclass = GetParam ();
+    T8_ASSERT (0 <= eclass && eclass < T8_ECLASS_COUNT);
+    Handle_Geom_Curve cad_curve;
+    TColgp_Array1OfPnt point_array (1, 3);
+    TopoDS_Shape shape;
+
+    /*  x--> u-parameter
+    *   
+    *  ----____       curve       ____----
+    *           ----_________----
+    * 
+    *  0               0.5              1
+    */
+
+    point_array (1) = gp_Pnt (0.0, 0.0, 0.0);
+    point_array (2) = gp_Pnt (0.5, -0.2, 0.0);
+    point_array (3) = gp_Pnt (1.0, 0.0, 0.0);
+
+    cad_curve = GeomAPI_PointsToBSpline (point_array).Curve ();
+    shape = BRepBuilderAPI_MakeEdge (cad_curve).Edge ();
+
+    geometry_cad = new t8_geometry_cad (3, shape, "cad curve dim=3");
+
+    geometry = geometry_cad;
+
+    t8_cmesh_init (&cmesh);
+    t8_cmesh_set_tree_class (cmesh, 0, eclass);
+  }
+
+  void
+  TearDown () override
+  {
+    t8_cmesh_destroy (&cmesh);
+  }
+  t8_cmesh_t cmesh;
+  t8_eclass_t eclass;
+  t8_geometry_cad *geometry_cad;
+  t8_geometry_c *geometry;
+
+  /* The arrays prescribe the linkage of the element. Edge 2 of the element is linked and the face is unlinked. */
+  int faces[1] = { 0 };
+  int edges[8] = { 0, 0, 1, 0, 0, 0, 0, 0 };
+};
+
+TEST_P (class_2d_element_curved_cad_curve, t8_check_2d_element_curved_cad_curve)
+{
+  /* Saving the corner vertices for the given element class. */
+  const int num_vertices = t8_eclass_num_vertices[eclass];
+  const double *vertices = &(t8_element_corner_ref_coords[eclass][0][0]);
+
+  t8_cmesh_set_tree_vertices (cmesh, 0, vertices, num_vertices);
+
+  /* Edges are parametrized in one parameter u. The arrays contain the parameters
+   * each vertex of the element has on the linked curve. */
+  double params[2] = { 0, 1 };
+
+  /* Passing of the attributes to the element */
+  t8_cmesh_set_attribute (cmesh, 0, t8_get_package_id (), T8_CMESH_CAD_FACE_ATTRIBUTE_KEY, faces, sizeof (int), 0);
+  t8_cmesh_set_attribute (cmesh, 0, t8_get_package_id (), T8_CMESH_CAD_EDGE_ATTRIBUTE_KEY, edges,
+                          2 * num_vertices * sizeof (int), 0);
+  t8_cmesh_set_attribute (cmesh, 0, t8_get_package_id (), T8_CMESH_CAD_EDGE_PARAMETERS_ATTRIBUTE_KEY + 2, params,
+                          2 * sizeof (double), 0);
+
+  /* Register the geometry */
+  t8_cmesh_register_geometry (cmesh, &geometry);
+  /* Commit the cmesh */
+  t8_cmesh_commit (cmesh, sc_MPI_COMM_WORLD);
+
+  double test_ref_coords_in[9] = { 0.0, 0.0, 0.0, 0.5, 0.0, 0.0, 1.0, 0.0, 0.0 };
+  double test_ref_coords_out[9] = { 0.0, 0.0, 0.0, 0.5, -0.2, 0.0, 1.0, 0.0, 0.0 };
```

Comment:
You could construct the curve with these coordinates, so you do not define them twice